code/Test_definitions/connected-network-type-subscriptions.feature

```diff
@@ -1,289 +1,488 @@
-    Given a valid subscription request body
-    And the request property "$.accessTokenType" is not "bearer"
+  @connected_network_type_subscriptions_400.6_create_subscription_with_invalid_access_token_type
+  Scenario: subscription creation with invalid token
+    Given use BaseURL
     When the request "createConnectedNetworkTypeSubscription" is sent
-    Then the response code is 400
-    And the response property "$.status" is 400
-    And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"
+    And a valid subscription request body
+    And "$.sink" is set to provided callbackUrl
+    And "$.sinkCredential.credentialType" = "ACCESSTOKEN"
+    And "$.sinkCredential.accessTokenType" <> "bearer"
+    And "$.sinkCredential.accessToken" is valued with a valid value
+    And "$.sinkCredential.accessTokenExpiresUtc" is valued with a valid value
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN"
     And the response property "$.message" contains a user friendly text
 
-  @connected_network_type_subscriptions_17_no_authorization_header_for_create_subscription
-  Scenario: subscription creation with invalid credentials
-    Given a valid subscription request body
-    And header "Authorization" token is set to invalid credentials
+##################
+# Error code 401
+##################
+
+  @connected_network_type_subscriptions_creation_401.1_no_authorization_header
+  Scenario: No Authorization header
+    Given the header "Authorization" is removed
+    And use BaseUrL
+    And the request body is set to a valid request body
     When the request "createConnectedNetworkTypeSubscription" is sent
-    Then the response code is 401
-    And the response property "$.status" is 401
+    Then the response property "$.status" is 401
     And the response property "$.code" is "UNAUTHENTICATED"
     And the response property "$.message" contains a user friendly text
 
-  @connected_network_type_subscriptions_18_expired_access_token_for_create_subscription
-  Scenario: Expired access token for create subscription
-    Given a valid subscription request body and header "Authorization" is expired
+  @connected_network_type_subscriptions_creation_401.2_expired_access_token
+  Scenario: Expired access token
+    Given the header "Authorization" is set to an expired access token
+    And use BaseUrL
+    And the request body is set to a valid request body
     When the request "createConnectedNetworkTypeSubscription" is sent
-    Then the response status code is 401
+    Then the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_creation_401.3_invalid_access_token
+  Scenario: Invalid access token
+    Given the header "Authorization" is set to an invalid access token
```

Comment:
We should maybe define "invalid" here. A token that is otherwise valid but only missing the required scope might be considered "invalid" by some, but would result in a 403 error.